File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitCharge/EngineerVisitChargeValidator.vb
Namespace Entity

Namespace EngineerVisitCharges

Public Class EngineerVisitChargeValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oEngineerVisitCharge As EngineerVisitCharge)

        'make sure that contact object is valid
        If oEngineerVisitCharge.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oEngineerVisitCharge.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitCharge/EngineerVisitChargeValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitCharges
    {
        // make sure that contact object is valid
        public class EngineerVisitChargeValidator : BaseValidator
        {
            public void Validate(EngineerVisitCharge oEngineerVisitCharge)
            {
                // make sure that contact object is valid
                if (oEngineerVisitCharge.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oEngineerVisitCharge.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}